packages/perseus-editor/src/item-extras-editor.tsx

```diff
@@ -20,159 +26,208 @@ class ItemExtrasEditor extends React.Component<Props> {
     static defaultProps: ItemExtrasProps = {
         calculator: false,
         chi2Table: false,
+        financialCalculator: false,
+        financialCalculatorMonthlyPayment: false,
+        financialCalculatorTotalAmount: false,
+        financialCalculatorTimeToPayOff: false,
         periodicTable: false,
         periodicTableWithKey: false,
         tTable: false,
         zTable: false,
     };
 
+    componentDidUpdate(prevProps: Readonly<Props>): void {
+        if (
+            prevProps.financialCalculator &&
+            !this.props.financialCalculatorMonthlyPayment &&
+            !this.props.financialCalculatorTotalAmount &&
+            !this.props.financialCalculatorTimeToPayOff
+        ) {
+            this.props.onChange({
+                financialCalculator: false,
+            });
+        }
+    }
+
     serialize: () => any = () => {
```

Comment:
**request**: Can you type this `serialize` method result as the `PerseusAnswerArea` type from `@khanacademy/perseus`, please? That will mean you also need to add these new options to that type, but that's a good thing as that type should be the source of truth for what options are available in the answer area.

Reply:
Can do! I wanted to type this but wasn't sure where it was used or how, didn't investigate. Always happy to improve types.

Reply:
@jeremywiebe it doesn't seem advisable to import types from a different package. Each package should export its own types, no? This is reinforcing some very difficult coupling between the packages.

It doesn't look like `PerseusAnswerArea` is used anywhere.

Reply:
OH I see now. I'll figure something out here.

Reply:
With regards to exporting/importing types. In this case, the `perseus-editor` package is used to build objects that are consumed by Perseus itself.  So in that case I do think it makes sense for the editors to use types from Perseus to ensure we are building the Perseus data objects correctly. 

In the future, I'd love to see these editors run the data through a validation that's exported from Perseus to guarantee the objects are valid before saving them (ie. not _just_ a type check but a runtime validation).